Eatery/Views/Eateries/HistogramViewController.swift

```diff
@@ -0,0 +1,285 @@
+//
+//  HistogramViewController.swift
+//  Eatery
+//
+//  Created by Ethan Fine on 4/26/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import UIKit
+
+// TODO: Create a protocol that encapsulates all Swipe Data and make HistogramViewController independent of Eatery variables
+class HistogramViewController: UIViewController {
+    
+    // MARK: Data
+    var swipeData: EaterySwipeData!
+    // hours in overflowable military time units
+    let hourRange = (startTime: 6, endTime: 3)
+    var overflowingEndTime: Int!
+    
+    // MARK: Views and view properties
+    var viewFrame: CGRect!
+    var axisView: UIView!
+    var axisTickViews = [UIView]()
+    var axisTickLabels = [UILabel]()
+    
+    var barWidth: Double!
+    var barContainerView: UIView!
+    var barViews = [UIView]()
+   
+    var expandedWaitTimeView: (barView: UIView, waitTimeView: UIView)?
+    
+    // MARK: Initializers
+    
+    init(frame: CGRect, swipeData: EaterySwipeData) {
+        super.init(nibName: nil, bundle: nil)
+        
+        self.viewFrame = frame
+        self.swipeData = swipeData
+        overflowingEndTime = (hourRange.endTime < hourRange.startTime) ? hourRange.endTime + 24 : hourRange.endTime
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func viewDidLoad() {
+        view.frame = viewFrame
+        
+        setUpBarViews()
+        setUpAxis()
+    }
+    
+    // MARK: Initial view set up
+    
+    private func setUpAxisLineView() {
+        let axisView = UIView()
+        axisView.backgroundColor = .inactive
+        view.addSubview(axisView)
+        
+        axisView.snp.makeConstraints { make in
+            make.height.equalTo(2)
+            make.bottom.equalToSuperview().inset(45)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview().inset(15);
+        }
+    }
+    
+    private func setUpAxis() {
+        setUpAxisLineView()
+        
+        let overflowingEndTime = (hourRange.endTime < hourRange.startTime) ? hourRange.endTime + 24 : hourRange.endTime
+        let tickCount: Int = ((overflowingEndTime - hourRange.startTime) / 3) + 1
+        
+        var x: Double = 0
+        for tick in 0..<tickCount {
```

Comment:
Can we find some way to do this more declaratively using constraints instead of merely using constraints as a way to set frames?

Reply:
not too sure about the intended design but could we use a `collectionView` here?